### PR TITLE
Bucket info and TTL functions for ddb

### PR DESCRIPTION
**Pull request by davecromberge**
https://github.com/dalmatinerdb/dalmatinerdb/issues/62
https://github.com/dalmatinerdb/dalmatinerdb/issues/63

### CODE REVIEW

File: apps/dalmatiner_db/src/dalmatiner_tcp.erl [hunk 1]
```diff
@@ -45,10 +45,6 @@ loop(Socket, Transport, State) ->
                     {ok, Bs} = metric:list(),
                     Transport:send(Socket, dproto_tcp:encode_metrics(Bs)),
```
Comment (elvisci) [inline apps/dalmatiner_db/src/dalmatiner_tcp.erl]: Remove the dynamic function call on line 46. Only modules that define callbacks should make dynamic calls.

File: apps/dalmatiner_db/src/dalmatiner_tcp.erl [hunk 2]
```diff
@@ -60,9 +56,17 @@ loop(Socket, Transport, State) ->
                     {ok, Ms} = metric:list(Bucket, Prefix),
                     Transport:send(Socket, dproto_tcp:encode_metrics(Ms)),
                     loop(Socket, Transport, State);
+                {info, Bucket} ->
+                    {ok, {Res, PPF, TTL}} = metric:bucket_info(Bucket),
+                    InfoBin = dproto_tcp:encode_bucket_info(Res, PPF, TTL),
+                    Transport:send(Socket, InfoBin),
```
Comment (elvisci) [inline apps/dalmatiner_db/src/dalmatiner_tcp.erl]: Remove the dynamic function call on line 62. Only modules that define callbacks should make dynamic calls.

File: apps/dalmatiner_db/src/dalmatiner_tcp.erl [hunk 3]
```diff
@@ -45,10 +45,6 @@ loop(Socket, Transport, State) ->
                     {ok, Bs} = metric:list(),
                     Transport:send(Socket, dproto_tcp:encode_metrics(Bs)),
                     loop(Socket, Transport, State);
-%%                {resolution, Bucket} ->
-%%                    Resolution = dalmatiner_opt:resolution(Bucket),
-%%                    Transport:send(Socket, <<Resolution:64/integer>>),
-%%                    loop(Socket, Transport, State);
                 {list, Bucket} ->
                     {ok, Ms} = metric:list(Bucket),
                     Transport:send(Socket, dproto_tcp:encode_metrics(Ms)),
```
Comment (elvisci) [inline apps/dalmatiner_db/src/dalmatiner_tcp.erl]: Remove the dynamic function call on line 50. Only modules that define callbacks should make dynamic calls.

File: apps/dalmatiner_db/src/dalmatiner_tcp.erl [hunk 4]
```diff
@@ -60,9 +56,17 @@ loop(Socket, Transport, State) ->
                     {ok, Ms} = metric:list(Bucket, Prefix),
                     Transport:send(Socket, dproto_tcp:encode_metrics(Ms)),
```
Comment (elvisci) [inline apps/dalmatiner_db/src/dalmatiner_tcp.erl]: Remove the dynamic function call on line 57. Only modules that define callbacks should make dynamic calls.

**Comment by Licenser** [comment]
Those elvis things should not be triggered

**Comment by davecromberge** [comment]
I'm guessing it is referring to Transport:send:
https://github.com/inaka/erlang_guidelines/blob/master/src/dyn_calls.erl 
Maybe it is implemented as a dyn function in cowboy?

**Comment by Licenser** [comment]
yes the transport:send thing is a known OK, the elvis.config should exclude that, I'll ignore those errors :)

**Pull request merged**